src/app/utils.ts

```diff
@@ -0,0 +1,9 @@
+export const generateSlug = (text: string) =>
+    text
+        .normalize("NFD") // Normalize Unicode characters to separate diacritics
+        .replace(/[\u0300-\u036f]/g, "") // Remove diacritical marks
+        .toLowerCase() // Convert to lowercase
+        .replace(/[^\p{L}\p{N}\s-]/gu, "") // Remove non-letter, non-number, non-space, non-dash characters
+        .replace(/\s+/g, "-") // Replace spaces with dashes
+        .substring(0, 200) // Limit to 200 chars
```

Comment:
I suppose we should substring in the first step, to avoid processing the whole text unncessarily.

Reply:
Makes sense
The only "issue" I can think with limiting at the start is that non-latin languages may end up with shorter slugs - pictographic languages may use more than 1 char per actual printed character - but that's hardly a problem here